### ISSUE TITLE
Alternative header syntax after code is "invalid"

### ISSUE THREAD

**Issue by pareeohnos** [RobertDober/earmark_parser#98]
I'm running into an error where Earmark is telling me my Markup is invalid but as far as I can tell, it's not. The below example can be used to produce the error:

```
markdown = """
This is a paragraph.

    This is some code.



Header 1
========

Header 2
--------
"""
```

This produces the following output:

```
{:error,
 [
   {"p", [],
    [
      {"a",
       [
         {"href",
          "https://raw.github.com/adamschwartz/github-markdown-kitchen-sink/master/README.md"}
       ], ["View raw (TEST.md)"], %{}}
    ], %{}},
   {"p", [], ["This is a paragraph."], %{}},
   {"pre", [], [{"code", [], ["This is a paragraph."], %{}}], %{}},
   {"p", [], ["Header 1"], %{}},
   {"p", [], [], %{}},
   {"h2", [], ["Header 2"], %{}}
 ], [{:warning, 10, "Unexpected line ========"}]}
```

If I try and parse just the header on its own, it works correctly

```
markdown = """
Header 1
=======
"""

EarmarkParser.as_ast(markdown)
{:ok, [{"h1", [], ["Header 1"], %{}}], []}
```

And if I remove the code block it also works just fine

```
markdown = """
This is a paragraph.


Header 1
========

Header 2
--------
"""

EarmarkParser.as_ast(markdown)
{:ok,
 [
   {"p", [], ["This is a paragraph."], %{}},
   {"h1", [], ["Header 1"], %{}},
   {"h2", [], ["Header 2"], %{}}
 ], []}
```

and also if I change the code block to use three back ticks instead of indentation, it also works

    markdown = """
    This is a paragraph.

    ```
    This is a some code
    ```
    
    
    Header 1
    ========
    
    Header 2
    --------
    """
    
    EarmarkParser.as_ast(markdown)
    {:ok,
     [
       {"p", [], ["This is a paragraph."], %{}},
       {"pre", [], [{"code", [], ["This is a paragraph."], %{}}], %{}},
       {"h1", [], ["Header 1"], %{}},
       {"h2", [], ["Header 2"], %{}}
     ], []}

Not sure why it's playing up only after an indented code block but that's the only time I can replicate it

**Comment by RobertDober** [comment]
I got lucky just released 1.4.21 with a fix

**Comment by pareeohnos** [comment]
oh wow that was fast! Thanks for this, I'll test it out
